src/components/manifold-subscription-details/manifold-subscription-details.tsx

```diff
@@ -14,21 +14,26 @@ import { SubscriptionDetails } from './components/SubscriptionDetails';
 })
 export class ManifoldSubscriptionDetails {
   @Element() el: HTMLElement;
-  @Prop() subscriptionId: string;
+  @Prop() subscriptionId?: string;
+  @Prop() planId?: string;
   @Prop() heading?: string;
   @Prop() isEditing?: boolean = false;
+  @Prop() preview?: boolean = false;
 
   // Can this be abstracted/optimized further?
   async componentWillLoad() {
     setConnection(await getManifoldConnection(this.el));
     setState('heading', this.heading);
     setState('subscriptionId', this.subscriptionId);
+    setState('preview', this.preview);
```

Comment:
Is there an advantage to storing this in state, as opposed to just using the prop?

What if `planId` updates in the client (React)? Do we have to `@Watch()` for changes here?

Reply:
I know that neither of these are likely to update, I’m just wondering if we can come up with rules for when it’s OK to duplicate props into Stencil Store, and when it’s not OK

Reply:
Passing everything through to the store lets us use this value without having to keep passing it down through every function. Yes, we would have to add a watch to update this if we want to allow that.
I've been leaving details like this out while we're still figuring out exactly what the component will look like so it doesn't complicate any refactoring that needs to happen. Ad you said though, this is unlikely to change, so I don't think it's essential functionality.